### PR TITLE
Feature/crowd 6 modify project description

### PR DESCRIPTION
**Pull request by wThomas84**

### CODE REVIEW

**Comment by CanardSauvage** [comment]
Checks failed, get me some green!

File: crowdsource-frontend/src/test/javascript/karma.conf.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ module.exports = function (config) {
             'src/main/resources/public/app/**/*.js',
             'src/test/javascript/**/*.js',
 
+            // Selective test execution -> comment line above uncomment lines below
```
Comment (CanardSauvage) [inline crowdsource-frontend/src/test/javascript/karma.conf.js]: Delete comment.

File: crowdsource-frontend/src/test/javascript/app/project/form/project-form.test.js [hunk 2]
```diff
@@ -14,25 +14,29 @@ describe('project form', function () {
         localStorage.clear(); // reset
 
         inject(function ($compile, $rootScope, $templateCache, $controller, _$httpBackend_, _$location_, Project, RemoteFormValidation) {
```
Comment (CanardSauvage) [inline crowdsource-frontend/src/test/javascript/app/project/form/project-form.test.js]: I still would prefer the setup to be more readable and recommend the usage of methodes.

File: crowdsource-frontend/src/main/scss/components/project/_project-details.scss [hunk 3]
```diff
@@ -271,6 +271,25 @@ $pd-icon-dimension: 24px;
 }
 
 /*button*/
+.pm-btn-simple--success {
```
Comment (CanardSauvage) [inline crowdsource-frontend/src/main/scss/components/project/_project-details.scss]: I don't like short forms like "pm", this makes the code less readable.

File: crowdsource-frontend/src/main/resources/public/app/project/details/project-pledging-form.js [hunk 4]
```diff
@@ -40,13 +44,11 @@ angular.module('crowdsource')
...
                 };
 
-                FinancingRound.reloadCurrentRound();
-
                 // to get the current user's budget
                 vm.user = Authentication.reloadUser();
 
                 vm.authorizedForPostRoundPledges = function () {
-                    var currentFinancingRound = FinancingRound.currentFinancingRound();
+                    var currentFinancingRound = vm.financingRound;
```
Comment (CanardSauvage) [inline crowdsource-frontend/src/main/resources/public/app/project/details/project-pledging-form.js]: I like this.

File: crowdsource-frontend/src/main/resources/public/app/project/details/project-details.js [hunk 5]
```diff
@@ -113,5 +119,25 @@ angular.module('crowdsource')
             || vm.project.status == 'DEFERRED' || vm.project.status == 'PROPOSED';
     };
 
-});
+    vm.editButtonVisibleForUser = function () {
+        var res = vm.auth.currentUser.hasRole("ADMIN") || Project.isCreator(vm.project, vm.auth.currentUser);
+        //console.log("EditBtn Visible: " + res);
```
Comment (CanardSauvage) [inline crowdsource-frontend/src/main/resources/public/app/project/details/project-details.js]: Remove comment.

**Comment by CanardSauvage** [comment]
This is so much again! *cry* But I like the small improvements and the quantity of the tests. The structure of some tests could be improved to make them more readable.

File: crowdsource-core/src/test/java/de/asideas/crowdsource/domain/model/ProjectEntityTest.java [hunk 6]
```diff
@@ -690,6 +692,142 @@ public void modifyStatus_settingToPublishedPossibleWhenAlreadyRejected() throws
...
+    @Test
+    public void masterdataModificationAllowed_shouldReturnTrueHavingStatusDeferred() throws Exception{
+        projectEntity.setFinancingRound(anActiveFinancingRound());
+        projectEntity.setStatus(ProjectStatus.DEFERRED);
+        assertThat(projectEntity.masterdataModificationAllowed(), is(true));
+    }
+
+    @Test
+    public void masterdataChanged_returnsTrueIfOneFieldChanges() throws Exception{
+        Project modifiedProject;
```
Comment (CanardSauvage) [inline crowdsource-core/src/test/java/de/asideas/crowdsource/domain/model/ProjectEntityTest.java]: This is not a good test.

A good test should 
* test the public interface of a class
* Be readable - tests are documentation and you read them a lot more then you edit them
* test only one thing (cf. GivenWhenThen)
* the name should tell what they test, what they change/modify and what the outcome should be (this is ok for this case ;) )

File: crowdsource-core/src/main/java/de/asideas/crowdsource/domain/presentation/user/ProjectCreator.java [hunk 7]
```diff
@@ -20,7 +19,7 @@
 
     public ProjectCreator(UserEntity user) {
         this.id = user.getId();
-        this.name = UserHelper.determineNameFromEmail(user.getEmail());
+        this.name = user.fullNameFromEmail();
```
Comment (CanardSauvage) [inline crowdsource-core/src/main/java/de/asideas/crowdsource/domain/presentation/user/ProjectCreator.java]: Way better.

File: crowdsource-frontend/src/test/javascript/karma.conf.js (hunk 1 above)
Comment (wThomas84) [inline crowdsource-frontend/src/test/javascript/karma.conf.js]: I'd like to leave it here, due to it's documentation for the developer

**Pull request merged**